packages/protocol/src/certificate/CertificateAuthority.ts

```diff
@@ -187,14 +267,155 @@ export class CertificateAuthority {
+            this.#icacState = {
+                certId: BigInt(certValues.icacCertId as bigint | number),
+                keyPair: PrivateKey(certValues.icacKeyPair as BinaryKeyPair),
+                keyIdentifier: certValues.icacKeyIdentifier as Bytes,
+                certBytes: certValues.icacCertBytes as Bytes,
+            };
+
+            logger.info("ICAC data detected in storage and loaded.");
+        } else if (this.#ica) {
+            logger.warn(
+                "ica=true in configuration but no valid ICAC data in storage. ICAC support will be unavailable.",
+            );
+        }
+    }
+
+    #buildStorageData(): CertificateAuthority.StorageData {
+        const data: CertificateAuthority.StorageData = {
+            rootCertId: this.#rootCertId,
+            rootKeyPair: this.#initializedRootKeyPair.keyPair,
+            rootKeyIdentifier: this.#initializedRootKeyIdentifier,
+            rootCertBytes: this.#initializedRootCertBytes,
+            nextCertificateId: this.#nextCertificateId,
+        };
+
+        if (this.#icacState) {
+            data.icacCertId = this.#icacState.certId;
+            data.icacKeyPair = this.#icacState.keyPair.keyPair;
+            data.icacKeyIdentifier = this.#icacState.keyIdentifier;
+            data.icacCertBytes = this.#icacState.certBytes;
+        }
+
+        return data;
+    }
+
+    #getSigningParameters(): {
+        issuer: { rcacId: bigint } | { icacId: bigint };
+        signingKey: PrivateKey;
+        authorityKeyId: Bytes;
+    } {
+        if (this.#signingCert === SigningCertType.Intermediate) {
+            if (!this.#icacState) {
+                throw new InternalError("ICAC signing requested but ICAC is not initialized");
+            }
+            return {
+                issuer: { icacId: this.#icacState.certId },
+                signingKey: this.#icacState.keyPair,
+                authorityKeyId: this.#icacState.keyIdentifier,
+            };
+        }
+
+        return {
+            issuer: { rcacId: this.#rootCertId },
+            signingKey: this.#initializedRootKeyPair,
+            authorityKeyId: this.#initializedRootKeyIdentifier,
+        };
+    }
+
+    get #initializedRootCertBytes() {
+        if (this.#rootCertBytes === undefined) {
+            throw new InternalError("Root certificate is not installed");
+        }
+        return this.#rootCertBytes;
+    }
+}
+
+interface IcacState {
+    certId: bigint;
+    keyPair: PrivateKey;
+    keyIdentifier: Bytes;
+    certBytes: Bytes;
 }
 
 export namespace CertificateAuthority {
+    export type PartialConfiguration = {
+        ica?: boolean;
+        signingCert?: SigningCertType | string;
+    };
+
     export type Configuration = {
         rootCertId: bigint;
         rootKeyPair: BinaryKeyPair;
         rootKeyIdentifier: Bytes;
         rootCertBytes: Bytes;
         nextCertificateId: bigint;
+        ica?: boolean;
+        signingCert?: SigningCertType | string;
```

Comment:
when handed in as configuration I would only use the enum type because this is then done by code, or?!

Reply:
Fixed, option was removed.